### PR TITLE
Use sentry_init fixture in tests instead of using Hub directly

### PR DESCRIPTION
**Pull request by alexmojaki**
@untitaker how's this?

Based on https://github.com/getsentry/sentry-python/pull/749#issuecomment-656827764

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -149,41 +156,41 @@ def test_proxy_httpsselect_bothenv_http(monkeypatch):
     assert client.transport._pool.proxy.scheme == "http"
 
 
-def test_simple_transport():
+def test_simple_transport(sentry_init):
     events = []
```
Comment (untitaker) [inline tests/test_client.py:160]: You do not need to pass in a transport if you're using sentry_init, just use the `capture_events` fixture instead which does the same thing.

Comment (alexmojaki) [reply tests/test_client.py:160]: I did that at first but then decided that wasn't in the spirit of the original test and the name `test_simple_transport`.

Comment (untitaker) [reply tests/test_client.py:160]: ah well you're right

**Comment by untitaker** [comment]
Thanks! Not entirely sure why project coverage is still there. I think I disabled it.

**Pull request merged**